packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/utils/select.js

```diff
@@ -12,8 +12,13 @@ function useSelect({ schema, componentFieldName }) {
   } = useCMEditViewDataManager();
 
   const mainField = useMemo(() => get(schema, ['settings', 'mainField'], 'id'), [schema]);
+  const nestedObjectTitle = (((((schema.layouts.edit || [])[0] || [])[0] || {}).metadatas || {}).mainField || {}).name;
+  const nestedObjectField = (((schema.layouts.edit || [])[0] || [])[0] || {}).name;
+  const displayValuePath = (mainField === 'id' && !!nestedObjectTitle)
+    ? [...componentFieldName.split('.'), nestedObjectField, nestedObjectTitle]
+    : [...componentFieldName.split('.'), mainField];
   const displayedValue = toString(
-    get(modifiedData, [...componentFieldName.split('.'), mainField], '')
+    get(modifiedData, displayValuePath, '')
```

Comment:
Hello @timbrandin,

Thanks for your contribution. IMO, for a better readability you can use `lodash/get` for `nestedObjectField` & `nestedObjectTitle` too.

```js
import get from 'lodash/get';
import toString from 'lodash/toString';

...

const defaultMainField = get(schema, ['settings', 'mainField'], 'id');
const mainField = get(
  schema,
  ['layouts', 'edit', 0, 0, 'metadatas', 'mainField', 'name'],
  defaultMainField
);
const nestedObjectField = get(schema, ['layouts', 'edit', 0, 0, 'name'], undefined);
const displayValuePath = nestedObjectField
  ? [...componentFieldName.split('.'), nestedObjectField, mainField]
  : [...componentFieldName.split('.'), defaultMainField];
const displayedValue = toString(get(modifiedData, displayValuePath, ''));
```

Reply:
Can you also remove the useless `useMemo` please